utils/tests/test_bundle_miner.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <unity/unity.h>
+
+#include "common/crypto/iss/normalize.h"
+#include "common/defs.h"
+#include "common/trinary/trit_tryte.h"
+#include "common/trinary/trits.h"
+#include "common/trinary/tryte.h"
+#include "utils/bundle_miner.h"
+
+#define SECURITY 2
+#define NORMALIZED_LENGTH (SECURITY * NORMALIZED_FRAGMENT_LENGTH)
+
+static int8_t MIN_ARRAY[NORMALIZED_LENGTH];
+static int8_t MAX_ARRAY[NORMALIZED_LENGTH];
+
+static void test_bundle_miner_probability_of_losing(void) {
+  float min_probability = bundle_miner_probability_of_losing(MIN_ARRAY, SECURITY);
+  float max_probability = bundle_miner_probability_of_losing(MAX_ARRAY, SECURITY);
+
+  TEST_ASSERT_EQUAL_FLOAT(min_probability, 0.0);
+  TEST_ASSERT_EQUAL_FLOAT(max_probability, 1.0);
+}
+
+static void test_bundle_miner_normalize_hash(void) {
+  tryte_t const* const bundle_trytes =
+      (tryte_t const* const) "QVXRKNRXFZIPFPREXRAPNHNSRFFQOWBGCAFZEGFCKDPDXRNVZQ9VJPQPPTFXKPVZVAIENQLETXRVSFKFO";
+  trit_t bundle_trits[HASH_LENGTH_TRIT];
+  byte_t bundle_expected[NORMALIZED_BUNDLE_LENGTH] = {
+      13, 13, 13, 13, 13, -2, -9, -3, 6,  -1, 9, -11, 6, -11, -9, 5,   -3, -9, 1,  -11, -13, 8,  -13, -8, -9, 6,  6,
```

Comment:
Can you keep the same space number between elements?

Reply:
No I can't, this is clang-format behavior.

Reply:
ah, I didn't notice that sorry.